storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -1273,6 +1292,10 @@ int main(int argc, char **argv) {
              ->probe_api_version_and_lookup(container_name)) {
       return EXIT_FAILURE;
     }
+    if (ec2_instance->get_is_ec2_instance_with_profile()) {
```

Comment:
can you also safeguard ec3_instance is not `nullptr`?

Reply:
We declare and create it binding to a new object:
```
std::shared_ptr<S3_ec2_instance> ec2_instance =
        std::make_shared<S3_ec2_instance>(&http_client);
```

It should never be nullptr as make_shared does not return nullptr in case of error.